test/e2e/node/pods.go

```diff
@@ -392,6 +393,14 @@ var _ = SIGDescribe("Pods Extended", func() {
 			ginkgo.By("updating the pod to have a negative TerminationGracePeriodSeconds")
 			pod.Spec.TerminationGracePeriodSeconds = utilpointer.Int64(-1)
 			_, err = podClient.PodInterface.Update(ctx, pod, metav1.UpdateOptions{})
+			if strings.Contains(err.Error(), genericregistry.OptimisticLockErrorMsg) {
+				// retry if the TerminationGracePeriodSeconds is overrided
+				// see more in https://github.com/kubernetes/kubernetes/pull/115606
+				pod, err = podClient.Get(ctx, pod.Name, metav1.GetOptions{})
+				framework.ExpectNoError(err, "failed to query for pod")
+				pod.Spec.TerminationGracePeriodSeconds = utilpointer.Int64(-1)
+				_, err = podClient.PodInterface.Update(ctx, pod, metav1.UpdateOptions{})
+			}
 			framework.ExpectNoError(err, "failed to update pod")
```

Comment:
don't hard-code a message check, wrap the `Get + modify + Update` lines in a RetryOnConflict block

Reply:
Updated. Thanks.